src/str.rs

```diff
@@ -63,6 +64,55 @@ pub trait ParallelString {
         Chars { chars: self.as_parallel_string() }
     }
 
+    /// Returns a parallel iterator over the characters of a string, with their positions.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rayon::prelude::*;
+    /// let min = "hello".par_char_indices().min_by_key(|&(_i, c)| c as i32);
+    /// assert_eq!(Some((1, 'e')), min);
+    /// ```
+    fn par_char_indices(&self) -> CharIndices {
+        CharIndices { chars: self.as_parallel_string() }
+    }
+
+    /// Returns a parallel iterator over the bytes of a string.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rayon::prelude::*;
+    /// let max = "hello".par_bytes().max();
+    /// assert_eq!(Some(b'o'), max);
+    /// ```
+    fn par_bytes(&self) -> Bytes {
+        let bytes = self.as_parallel_string().as_bytes();
+        Bytes { inner: bytes.par_iter().cloned() }
+    }
+
+    /// Returns a parallel iterator over a string encoded as UTF-16.
+    ///
+    /// Note that surrogate pairs (for codepoints greater than `U+FFFF`) are
+    /// produced as separate items, but will not be split across threads.
```

Comment:
This feature of not splitting pairs across threads came naturally from the way the string is split.  But I just realized that I didn't make a similar guarantee on `par_bytes` -- perhaps we should?  We could only do that as an unindexed producer though, to avoid arbitrary splits.

Reply:
I decided I am going to make that guarantee for `par_bytes()` too.  I think this is a more interesting feature to provide.  If someone really wanted the indexed `Bytes`, they can still use a raw `s.as_bytes().par_iter().cloned()` to get that effect.